src/containerbuddy/ips.go

```diff
@@ -0,0 +1,253 @@
+}
+
+var (
+	ifaceSpec = regexp.MustCompile(`^(?P<Name>\w+)(?:(?:\[(?P<Index>\d+)\])|(?::(?P<Version>inet6?)))?$`)
+)
+
+func parseInterfaceSpec(spec string) (interfaceSpec, error) {
+	if spec == "inet" {
+		return interfaceSpec{IPv6: false, Name: "*"}, nil
+	}
+	if spec == "inet6" {
+		return interfaceSpec{IPv6: true, Name: "*"}, nil
+	}
+
+	if match := ifaceSpec.FindStringSubmatch(spec); match != nil {
+		name := match[1]
+		index := match[2]
+		inet := match[3]
+		if index != "" {
+			i, _ := strconv.Atoi(index)
+			if i > 0 {
+				return interfaceSpec{Spec: spec, Name: name, Index: i}, nil
+			}
+			return interfaceSpec{Spec: spec}, fmt.Errorf("Unable to parse interface spec: %s. Index should be > 0", spec)
+		}
+		if inet != "" {
+			if inet == "inet" {
+				return interfaceSpec{Spec: spec, Name: name, IPv6: false}, nil
+			}
+			return interfaceSpec{Spec: spec, Name: name, IPv6: true}, nil
+		}
+		return interfaceSpec{Spec: spec, Name: name, IPv6: false}, nil
+	}
+	if _, net, err := net.ParseCIDR(spec); err == nil {
+		return interfaceSpec{Spec: spec, Network: net}, nil
+	}
+	return interfaceSpec{Spec: spec}, fmt.Errorf("Unable to parse interface spec: %s", spec)
+}
+
+type interfaceIP struct {
+	Name string
+	IP   net.IP
+}
+
+func (iip interfaceIP) To16() net.IP {
+	return iip.IP.To16()
+}
+
+func (iip interfaceIP) To4() net.IP {
+	return iip.IP.To4()
+}
+
+func (iip interfaceIP) IsIPv4() bool {
+	return iip.To4() != nil
+}
+
+func (iip interfaceIP) IPString() string {
+	if v4 := iip.To4(); v4 != nil {
+		return v4.String()
+	}
+	return iip.IP.String()
+}
+
+func (iip interfaceIP) String() string {
+	return fmt.Sprintf("%s:%s", iip.Name, iip.IPString())
+}
+
+// Queries the network interfaces on the running machine and returns a list
+// of IPs for each interface. Currently, this only returns IPv4 addresses.
+func getinterfaceIPs(interfaces []net.Interface) ([]interfaceIP, error) {
+	var ifaceIPs []interfaceIP
+	var errors []string
+
+	for _, intf := range interfaces {
+		ipAddrs, addrErr := intf.Addrs()
+
+		if addrErr != nil {
+			errors = append(errors, addrErr.Error())
+			continue
+		}
+
+		for _, ipAddr := range ipAddrs {
+			// Addresses some times come in the form "192.168.100.1/24 2001:DB8::/48"
+			// so they must be split on whitespace
+			for _, splitIP := range strings.Split(ipAddr.String(), " ") {
+				ip, _, err := net.ParseCIDR(splitIP)
+				if err != nil {
+					errors = append(errors, err.Error())
+					continue
+				}
+				intfIP := interfaceIP{Name: intf.Name, IP: ip}
+				ifaceIPs = append(ifaceIPs, intfIP)
+			}
+		}
+	}
+
+	// Stable Sort the interface IPs so that selecting the correct IP in GetIP
+	// can be consistent
+	sort.Stable(ByInterfaceThenIP(ifaceIPs))
+
+	/* If we had any errors parsing interfaces, we accumulate them all and
+	 * then return them so that the caller can decide what they want to do. */
+	if len(errors) > 0 {
+		err := fmt.Errorf(strings.Join(errors, "\n"))
+		println(err.Error())
```

Comment:
I carried this over from this:
https://github.com/joyent/containerbuddy/blob/master/src/containerbuddy/config.go#L467

However, I do agree it should be logged. I'll change it in both places.